Core/src/EnvisionException.cpp

```diff
@@ -39,13 +39,12 @@ bool& EnvisionException::assertOnThrow()
 }
 
 EnvisionException::EnvisionException()
-{
-	if (assertOnThrow()) Q_ASSERT_X(false, "Exception thrown", "");
-}
+	: EnvisionException("")
+{}
 
-EnvisionException::EnvisionException(const QString& message) : msg_{message}
+EnvisionException::EnvisionException(const QString& message, bool throwInDebugMode) : msg_{message}
```

Comment:
Why not just ```assertInDebugMode``` and make it true by default? That was my intention I just named it wrongly.?